controllers/templatesync/template_sync.go

```diff
@@ -523,11 +531,89 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		policySystemErrorsCounter.WithLabelValues(instance.Name, "", "client-error").Inc()
 	}
 
+	err = r.cleanUpExcessTemplates(ctx, dClient, *instance, parentTemplates)
+	if err != nil {
+		resultError = err
+		reqLogger.Error(resultError, "Error cleaning up templates")
+	}
+
 	reqLogger.Info("Completed the reconciliation")
 
 	return reconcile.Result{}, resultError
 }
 
+// cleanUpExcessTemplates compares existing policy templates on the cluster to those contained in the policy,
+// and deletes those that have been renamed or removed from the parent policy
+func (r *PolicyReconciler) cleanUpExcessTemplates(
+	ctx context.Context,
+	dClient dynamic.Interface,
+	instance policiesv1.Policy,
+	parentTemplates []string,
+) error {
+	// build list of policy template resources to check
+	tmplCRDs, err := dClient.Resource(schema.GroupVersionResource{
+		Group:    "apiextensions.k8s.io",
+		Version:  "v1",
+		Resource: "customresourcedefinitions",
+	}).List(ctx, metav1.ListOptions{
+		LabelSelector: "policy.open-cluster-management.io/policy-type=template",
+	})
+	if err != nil {
+		return fmt.Errorf("error getting policy template CRDs: %w", err)
+	}
+
+	tmplGVRs := []schema.GroupVersionResource{}
+
+	for _, crd := range tmplCRDs.Items {
+		versions, ok := crd.Object["spec"].(map[string]interface{})["versions"].([]interface{})
+		if !ok {
+			return fmt.Errorf("template CRD is formatted incorrectly: no versions found")
+		}
+
+		for _, version := range versions {
+			tmplGVRs = append(tmplGVRs, schema.GroupVersionResource{
```

Comment:
You don't need to select every version. You just need to pick one that is supported by the CRD (i.e. the latest one).